File: Java/Easy/WrongAnswer/take-gifts-from-the-richest-pile_4.java
class Solution {
    public long pickGifts(int[] gifts, int k) {
        
        PriorityQueue<Integer> q = new PriorityQueue<>(Collections.reverseOrder());
        for (int i = 0; i < gifts.length; i++) {
            q.add(gifts[i]);
        }
        while (k > 0) {
            int max = q.poll();
            q.add(max - (int)Math.floor(Math.sqrt(max)));
            k--;
        }
        long sum = 0;
        while (!q.isEmpty()) {
            sum += q.poll();
        }
        return sum;
    }
}

File: C++/Easy/Accepted/take-gifts-from-the-richest-pile_4.cpp
class Solution {
public:
    long long pickGifts(vector<int>& gifts, int k) {
        
        priority_queue<int> pq;
        for (auto& g : gifts) {
            pq.push(g);
        }
        while (k--) {
            auto g = pq.top();
            pq.pop();
            g = sqrt(g);
            pq.push(g);
        }
        long long ans = 0;
        while (!pq.empty()) {
            ans += pq.top();
            pq.pop();
        }
        return ans;
    }
};